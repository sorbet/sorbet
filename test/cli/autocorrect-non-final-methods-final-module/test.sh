#!/usr/bin/env bash

cwd="$(pwd)"
infile="$cwd/test/cli/autocorrect-non-final-methods-final-module/autocorrect-non-final-methods-final-module.rb"

tmp="$(mktemp -d)"

cp "$infile" "$tmp"

cd "$tmp" || exit 1
# Suppress all output because the ordering of errors about non-final methods is
# dependent on hash table iteration order.
if "$cwd/main/sorbet" --silence-dev-message -a autocorrect-non-final-methods-final-module.rb 1>/dev/null 2>&1; then
  echo "Expected to fail!"
  exit 1
fi

echo
echo --------------------------------------------------------------------------
echo

# Cat the file, to make sure that the autocorrect applied
cat autocorrect-non-final-methods-final-module.rb

rm autocorrect-non-final-methods-final-module.rb

rmdir "$tmp"
