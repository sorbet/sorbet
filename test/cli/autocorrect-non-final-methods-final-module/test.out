
--------------------------------------------------------------------------

# typed: true

class FinalClass
  extend T::Sig
  extend T::Helpers

  final!

  sig(:final) {returns(T.untyped)}
  def needs_autocorrect; end
  def no_sig_suggestion1; end

  sig(:final) {returns(T.untyped)}
  def needs_autocorrect_multiline
    nil
  end
  def no_sig_suggestion2; end

  sig(:final) do
    returns(T.untyped)
  end
  def needs_autocorrect_multiline_sig; end
  def no_sig_suggestion3; end

  sig(:final) do
    returns(T.untyped)
  end
  def needs_autocorrect_multiline_multiline_sig
    nil
  end
  def no_sig_suggestion4; end

  def no_sig_suggestion_multiline
    nil
  end

  attr_reader :no_sig_attr

  sig(:final) {returns(T.untyped)}
  attr_reader :sig_attr
  def no_sig_suggestion5; end

  sig(:final) {params(sig_writer: T.untyped).returns(T.untyped)}
  attr_writer :sig_writer
  def no_sig_suggestion6; end

  sig(:final) {returns(T.untyped)}
  attr_accessor :sig_accessor
  def no_sig_suggestion7; end

  sig(:final) {returns(T.untyped)}
  attr_reader :sig_attr_multi1, :sig_attr_multi2
  def no_sig_suggestion8; end

  sig(:final) {returns T.untyped}
  attr_reader :sig_attr_final1
  def no_sig_suggestion9; end

  sig(:final) do
    returns T.untyped
  end
  attr_reader :sig_attr_final2
  def no_sig_suggestion10; end
end
